tests/sentry/sentry_apps/api/endpoints/test_installation_service_hook_projects.py

```diff
@@ -0,0 +1,98 @@
+from django.urls import reverse
+
+from sentry.sentry_apps.models.servicehook import ServiceHook, ServiceHookProject
+from sentry.testutils.cases import APITestCase
+
+
+class SentryAppInstallationServiceHookProjectsEndpointTest(APITestCase):
+    def setUp(self):
+        self.user = self.create_user(email="boop@example.com")
+        self.org = self.create_organization(owner=self.user)
+        self.project = self.create_project(organization=self.org)
+        self.project2 = self.create_project(organization=self.org)
+
+        self.sentry_app = self.create_sentry_app(
+            name="Testin", organization=self.org, webhook_url="https://example.com"
+        )
+
+        self.install = self.create_sentry_app_installation(
+            organization=self.org, slug=self.sentry_app.slug, user=self.user
+        )
+
+        self.service_hook = ServiceHook.objects.get(
+            installation_id=self.install.id,
+        )
+
+        self.url = reverse(
+            "sentry-api-0-sentry-app-installation-service-hook-projects", args=[self.install.uuid]
+        )
+
+    def test_get_service_hook_projects(self):
+        self.login_as(user=self.user)
+
+        # Create a service hook project
+        ServiceHookProject.objects.create(
+            project_id=self.project.id, service_hook_id=self.service_hook.id
+        )
+
+        response = self.client.get(self.url, format="json")
+        assert response.status_code == 200
+        assert len(response.data) == 1
+        assert response.data[0]["project_id"] == self.project.id
+
+    def test_put_service_hook_projects(self):
```

Comment:
I'd like to see a test that proves the update works. So we have two projects, then we send an update that
 - causes one project to be removed
 - keeps one project
 - adds one new project

Reply:
done